### ISSUE TITLE
Build fails with --enable-werror on some non-x86 architecture

### ISSUE THREAD

**Issue by ggardet** [qpdf/qpdf#604]
When `--enable-werror ` is passed to configure script, the build fail on aarch64, armv6/7 and others.
The error log is about signed vs unsigned char:
```
[   78s] libqpdf/QPDFTokenizer.cc: In member function 'void QPDFTokenizer::findEI(PointerHolder<InputSource>)':
[   78s] libqpdf/QPDFTokenizer.cc:722:38: error: conversion to 'signed char' from 'char' may change the sign of the result [-Werror=sign-conversion]
[   78s]   722 |                     signed char ch = *iter;
[   78s]       |                                      ^~~~~
[   78s] libqpdf/QPDFTokenizer.cc:732:54: error: conversion to 'char' from 'signed char' may change the sign of the result [-Werror=sign-conversion]
[   78s]   732 |                     else if ((ch < 32) && (! isSpace(ch)))
[   78s]       |                                                      ^~
[   78s] cc1plus: all warnings being treated as errors
```

**Comment by jberkenbilt** [comment]
Thanks for bringing this to my attention. This is probably an issue with some platforms using unsigned char by default. I know powerpc64 was like that. I'll find one of the platforms you mentioned and make sure the build is clean with -Werror before the next release.

**Comment by jberkenbilt** [comment]
@ggardet I tagged you in a pull request, but if it's easier, here's a proposed fix. If this doesn't do it, I will go to the trouble of reproducing it, but a quick perusal of the code makes me think this is probably the only case that's wrong.

```patch
commit d7ab3b42b3c7278ddf08c22676cd544ceb96c51c
Author: Jay Berkenbilt <ejb@ql.org>
Date:   Mon Jan 10 11:52:07 2022 -0500

    Fix signed/unsigned char warning (fixes #604)

diff --git a/libqpdf/QPDFTokenizer.cc b/libqpdf/QPDFTokenizer.cc
index 4217575c..f4439fd1 100644
--- a/libqpdf/QPDFTokenizer.cc
+++ b/libqpdf/QPDFTokenizer.cc
@@ -719,7 +719,7 @@ QPDFTokenizer::findEI(PointerHolder<InputSource> input)
                 for (std::string::iterator iter = value.begin();
                      iter != value.end(); ++iter)
                 {
-                    signed char ch = *iter;
+                    char ch = *iter;
                     if (((ch >= 'a') && (ch <= 'z')) ||
                         ((ch >= 'A') && (ch <= 'Z')) ||
                         (ch == '*'))
@@ -729,10 +729,11 @@ QPDFTokenizer::findEI(PointerHolder<InputSource> input)
                         // alphabetic characters.
                         found_alpha = true;
                     }
-                    else if ((ch < 32) && (! isSpace(ch)))
+                    else if ((static_cast<signed char>(ch) < 32) &&
+                             (! isSpace(ch)))
                     {
-                        // ch is signed, so characters outside of
-                        // 7-bit will be < 0.
+                        // Compare ch as a signed char so characters
+                        // outside of 7-bit will be < 0.
                         found_non_printable = true;
                         break;
                     }
```
